### PR TITLE
HackStudio: Ask to create a non-existent directory when creating a project 

### PR DESCRIPTION
**Pull request by cbyrneee**
Previously, if the parent directory didn't exist when making a project, it would say that the creation directory was 'invalid' which isn't necessarily true.

This pull request will prompt the user, asking if they would like to create the parent directory when creating a project, instead of treating it as an 'invalid' directory.

### CODE REVIEW

**Comment by cbyrneee** [comment]
The failure on Azure x86_64 is not related to this pull request

**Comment by cbyrneee** [comment]
Incase anyone is wondering, those 2 force pushes were to fix the CI behaving weirdly :^)

File: Userland/DevTools/HackStudio/Dialogs/NewProjectDialog.cpp [hunk 1]
```diff
@@ -170,7 +170,7 @@ Optional<String> NewProjectDialog::get_available_project_name()
     return {};
 }
 
-Optional<String> NewProjectDialog::get_project_full_path()
+Optional<LexicalPath> NewProjectDialog::get_project_full_path()
```
Comment (itamar8910) [inline Userland/DevTools/HackStudio/Dialogs/NewProjectDialog.cpp:170]: `LexicalPath` is meant to be used as a utility class for parsing paths, and not really as a container for passing paths around.
Let's go back to simply using `String` here.

**Review by itamar8910 — CHANGES_REQUESTED** [review]

File: Userland/DevTools/HackStudio/Dialogs/NewProjectDialog.cpp (hunk 1 above)
Comment (cbyrneee) [reply Userland/DevTools/HackStudio/Dialogs/NewProjectDialog.cpp:170]: It is used for using ``parent()`` later, should I just construct it from a string when that's required?

**Review by itamar8910 — APPROVED** [review]

**Pull request merged**